src/ImageSharp/Advanced/AdvancedImageExtensions.cs

```diff
@@ -76,8 +74,8 @@ public static void AcceptVisitor(this Image source, IImageVisitor visitor)
         /// <param name="source">The source.</param>
         /// <param name="visitor">The visitor.</param>
         /// <returns>A  <see cref="Task"/> representing the asynchronous operation.</returns>
-        public static Task AcceptVisitorAsync(this Image source, IImageVisitorAsync visitor)
-            => source.AcceptAsync(visitor);
+        public static async Task AcceptVisitorAsync(this Image source, IImageVisitorAsync visitor)
+            => await source.AcceptAsync(visitor).ConfigureAwait(false);
```

Comment:
This isn't required... the ConfigureAwait only applies inside this `async Task` scope so if you don't await then it doesn't help..

Basically doing this will create the async state machine with no benefit to the caller.

Reply:
What about `SaveAsync`?

Reply:
You mean delete `ConfigureAwait(false)`?

Reply:
unless we 'need' to await (because of being in a wrapped using statements or require processing the result etc) then you should not await a task and pass the Task directly down stream, even out of your public api because it'll not help and only hinder due to allocations etc

So in conclusion revert this change and leave it alone, we also shouldn't even call it in the SaveAsync and should just pass everything along to the user.

Reply:
This also applies to most of the other places you changed in the commit

Reply:
I don't know. According to the general guidance the changes I've added are correct.

>This leads to the general guidance of: if you’re writing general-purpose library code, use ConfigureAwait(false). 

https://devblogs.microsoft.com/dotnet/configureawait-faq/

Reply:
The only place I would generally not bother with async/await is the result of `Task.FromResult` to remove the overhead.

Reply:
that only applies if you calling `await` and in all these places we don't need to call await so we don't need to call `ConfigureAwait(false)`

Reply:
> This leads to the general guidance of: if you’re writing general-purpose library code, use ConfigureAwait(false). This is why, for example, you’ll see every (or almost every) await in the .NET Core runtime libraries using ConfigureAwait(false) on every await; with a few exceptions, in cases where it doesn’t it’s very likely a bug to be fixed. For example, this PR fixed a missing ConfigureAwait(false) call in HttpClient.

Next sentence

Reply:
Ah... right, I'm being slow here. Will revert.